### PR TITLE
Add missing examples on Vec iter types

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

File: library/alloc/src/vec.rs [hunk 1]
```diff
@@ -2797,6 +2797,13 @@ where
 ///
 /// This `struct` is created by the `into_iter` method on [`Vec`] (provided
 /// by the [`IntoIterator`] trait).
+///
+/// # Example
+///
+/// ```
+/// let v = vec![0, 1, 2];
+/// let iter = v.into_iter();
```
Comment (jyn514) [inline library/alloc/src/vec.rs:2805]: Maybe say explicitly these are the same type?
```suggestion
/// let iter: std::vec::IntoIter<_> = v.into_iter();
```

File: library/alloc/src/vec.rs [hunk 2]
```diff
@@ -3040,6 +3047,13 @@ impl<T> AsIntoIter for IntoIter<T> {
 ///
 /// This `struct` is created by [`Vec::drain`].
 /// See its documentation for more.
+///
+/// # Example
+///
+/// ```
+/// let mut v = vec![0, 1, 2];
+/// let iter = v.drain(..);
```
Comment (jyn514) [inline library/alloc/src/vec.rs:3055]:
```suggestion
/// let iter: std::vec::Drain<_> = v.drain(..);
```

File: library/alloc/src/vec.rs [hunk 3]
```diff
@@ -3169,6 +3183,14 @@ impl<T> FusedIterator for Drain<'_, T> {}
 ///
 /// This struct is created by [`Vec::splice()`].
 /// See its documentation for more.
+///
+/// # Example
+///
+/// ```
+/// let v = vec![0, 1, 2];
+/// let new = [7, 8];
+/// let iter = v.splice(1.., new.iter().cloned());
```
Comment (jyn514) [inline library/alloc/src/vec.rs:3192]:
```suggestion
/// let iter: std::vec::Splice<_> = v.splice(1.., new.iter().cloned());
```

File: library/alloc/src/vec.rs [hunk 4]
```diff
@@ -3285,6 +3307,15 @@ impl<T> Drain<'_, T> {
...
 /// This struct is created by [`Vec::drain_filter`].
 /// See its documentation for more.
+///
+/// # Example
+///
+/// ```
+/// #![feature(drain_filter)]
+///
+/// let v = vec![0, 1, 2];
+/// let iter = v.drain_filter(|x| *x % 2 == 0);
```
Comment (jyn514) [inline library/alloc/src/vec.rs:3317]:
```suggestion
/// let iter: std::vec::DrainFilter<_, _> = v.drain_filter(|x| *x % 2 == 0);
```

File: library/alloc/src/vec.rs (hunk 1 above)
Comment (Dylan-DPC) [reply library/alloc/src/vec.rs:2805]: Not sure what value it adds to the example, Given it's a vector it's assumed to be items of the same type or I missed something?

Comment (jyn514) [reply library/alloc/src/vec.rs:2805]: This is documentation on `IntoIter`, so this tests that `.into_iter()` does in fact return `IntoIter`.

**Comment by pickfire** [comment]
Wouldn't this duplicate the documentation on the original function? Now there are two examples, one in the function and one for the struct. I also recall there is another patch for this. IMHO not worth it since it may create redundancy in docs as the original function already contains example.

**Comment by GuillaumeGomez** [comment]
@pickfire And like I said in the other PR, the examples here are focused on how to get this type, not how to use it (which is showed in the original function). All public items should have code examples.

**Review by pickfire — COMMENTED** [review]
Looks good to me.

**Review by pickfire — APPROVED** [review]
Looks good to me.

**Pull request merged**